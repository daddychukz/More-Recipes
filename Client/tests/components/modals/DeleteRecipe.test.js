import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import DeleteRecipe from '../../../src/components/modals/DeleteRecipeModal';


const props = {
  onClick: sinon.spy(() => new Promise((cb) => {
    cb();
  }))
};

const shallowWrapper = shallow(<DeleteRecipe {...props} />);

describe('Delete Recipe Modal test suite', () => {
  it('renders delete recipe modal without crashing', () => {
    expect(shallowWrapper).toBeDefined();
  });

  it('should match component snapshot', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});
