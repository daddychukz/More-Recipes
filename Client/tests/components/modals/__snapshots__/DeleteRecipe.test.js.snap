// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Recipe Modal test suite should match component snapshot 1`] = `
<div
  className="modal fade text-dark"
  id="delete-recipe"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Message
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <p>
           Do you realy want to delete this Recipe?
        </p>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-primary"
          data-dismiss="modal"
          id="delete"
          onClick={[Function]}
          type="button"
        >
          Yes Delete!
        </button>
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;
